src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -260,10 +268,16 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
       // Create the bucket in leo's google project and populate with initialization files.
       // ACLs are granted so the cluster service account can access the bucket at initialization time.
       initBucketPath <- initializeBucket(userEmail, googleProject, clusterName, initBucketName, clusterRequest, serviceAccountInfo, serviceAccountKeyOpt)
+      //Get list of groups to give staging bucket access to
+      groupstagingBucketAcl <- serviceAccountProvider.listGroupsStagingBucketReaders(userEmail)
+      userstagingBucketAcl <- serviceAccountProvider.listUsersStagingBucketReaders(userEmail)
+      // Create the staging bucket
+      stagingBucketPath <- gdDAO.createStagingBucket(googleProject, googleProject, stagingBucketName, serviceAccountInfo, groupstagingBucketAcl, userstagingBucketAcl)
       // Once the bucket is ready, build the cluster
-      cluster <- gdDAO.createCluster(userEmail, googleProject, clusterName, clusterRequest, initBucketName, serviceAccountInfo).andThen { case Failure(_) =>
+      cluster <- gdDAO.createCluster(userEmail, googleProject, clusterName, clusterRequest, initBucketName, serviceAccountInfo, stagingBucketPath).andThen { case Failure(_) =>
         // If cluster creation fails, delete the init bucket asynchronously
         gdDAO.deleteBucket(googleProject, initBucketName)
+        gdDAO.deleteBucket(googleProject, stagingBucketName)
```

Comment:
I don't think we want to delete the staging bucket if cluster creation fails, so we can access the error logs?

We probably only want to delete the staging bucket once the cluster has been explicitly deleted by the user.

Reply:
this is the third place where we do this cleanup...